tsl/src/continuous_aggs/create.c

```diff
@@ -1751,44 +1751,59 @@ cagg_create(ViewStmt *stmt, Query *panquery, CAggTimebucketInfo *origquery_ht,
 	return;
 }
 
-/* entry point for creating continuous aggregate view
- * step 1 : validate query
- * step 2: create underlying tables and views
- */
-DDLResult
-tsl_process_continuous_agg_viewstmt(ViewStmt *stmt, const char *query_string, void *pstmt,
-									WithClauseResult *with_clause_options)
+static bool
+continuous_agg_process_create_stmt(CreateTableAsStmt *stmt, const char *query_string, void *pstmt,
+								   WithClauseResult *with_clause_options)
 {
-	Query *query = NULL;
 	CAggTimebucketInfo timebucket_exprinfo;
 	Oid nspid;
-	PlannedStmt *pstmt_info = (PlannedStmt *) pstmt;
-	RawStmt *rawstmt = NULL;
-	/* we have a continuous aggregate query. convert to Query structure
-	 */
-	rawstmt = makeNode(RawStmt);
-	rawstmt->stmt = (Node *) copyObject(stmt->query);
-	rawstmt->stmt_location = pstmt_info->stmt_location;
-	rawstmt->stmt_len = pstmt_info->stmt_len;
-	query = parse_analyze(rawstmt, query_string, NULL, 0, NULL);
+	ViewStmt viewstmt = {
+		.type = T_ViewStmt,
+		.view = stmt->into->rel,
+		.query = stmt->into->viewQuery,
+		.options = stmt->into->options,
+		.aliases = stmt->into->colNames,
+	};
 
-	nspid = RangeVarGetCreationNamespace(stmt->view);
-	if (get_relname_relid(stmt->view->relname, nspid))
+	nspid = RangeVarGetCreationNamespace(stmt->into->rel);
+
+	if (get_relname_relid(stmt->into->rel->relname, nspid))
 	{
 		ereport(ERROR,
 				(errcode(ERRCODE_DUPLICATE_TABLE),
-				 errmsg("continuous aggregate query \"%s\" already exists", stmt->view->relname),
+				 errmsg("continuous aggregate query \"%s\" already exists",
+						stmt->into->rel->relname),
 				 errhint("drop and recreate if needed.  This will drop the underlying "
 						 "materialization")));
```

Comment:
Since you are changing this code anyway, we should also fix this hint since it is not following error guidelines. I actually don't think the hint adds any value (it is self-evident) so it can be removed. Otherwise I suggest:

```suggestion
				 errmsg("continuous aggregate \"%s\" already exists",
						stmt->into->rel->relname),
				 errhint("Drop or rename the existing continuous aggregate first or use another name.")));
```